src/array.js

```diff
@@ -49,10 +49,13 @@ TransisArray.of = function() {
 // Public: Creates a new `Transis.Array` from the given array-like object. Useful for converting a
 // regular array to a `Transis.Array`.
 //
-// a - An array-like object.
+// a     - An array-like object.
+// mapFn - Map function to call on every element of the array (optional).
 //
 // Returns a new `Transis.Array`.
-TransisArray.from = function(a) { return TransisArray.of.apply(null, a); };
+TransisArray.from = function(a, mapFn) {
+  return TransisArray.of.apply(null, mapFn ? a.map(a => mapFn(a)) : a);
```

Comment:
This assumes that the passed in argument has a `map` method. I don't think thats a valid assumption to make here. For example, previously you could pass an `Arguments` object to `TransisArray.from` and it would work, but if you do that now with a map function I think it would break.

The contract is that you pass in an "array-like" object, which really just means it has a `length` property. To be safe you should perform the map with a `for` loop instead.